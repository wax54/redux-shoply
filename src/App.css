.App {
  text-align: center;
  display: flex;
  flex-direction: column;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 6vh;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}
.App-header > a {
  margin: 10px;
  padding: 5px;
  text-decoration: none;
  color: white;
  background-color: lightgray;
  border-radius: 15px;
}

.App-link {
  color: #61dafb;
}

.App-main {
  background-color: #ddd;
  display: flex;
  flex: 1;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
