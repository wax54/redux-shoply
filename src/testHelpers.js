import { render, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router';

import { Provider } from 'react-redux';
import { createStore } from 'redux';
import rootReducer from './redux-setup/rootReducer';
const store = createStore(rootReducer);

const DEFAULT_OPTIONS = {
    initialEntries: ['/']
}

function preRender(options={}) {
    const newOptions = {
        initialEntries: 
            [...(options.initialEntries || []), ...DEFAULT_OPTIONS.initialEntries],
        context: 
            { ...DEFAULT_OPTIONS.context, ...(options.context || {}) }
    }
    //console.log('options: ', options, 'newOptions', newOptions);
    return newOptions;
}


function renderWithRedux(Component, options) {
    options = preRender(options);
    return render(
        <Provider store={store}>
            <MemoryRouter initialEntries={options.initialEntries}>
                    { Component }
            </MemoryRouter>
        </Provider>
    );
}

async function renderWithReduxAndWait(Component, options) {
    options = preRender(options);
    return await waitFor(() => render(
        <Provider store={store}>
            <MemoryRouter initialEntries={options.initialEntries}>
                {Component}
            </MemoryRouter>
        </Provider>
    ));
}



export { renderWithRedux, renderWithReduxAndWait}