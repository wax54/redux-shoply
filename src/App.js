import { Switch, Route, Redirect, Link} from 'react-router-dom';
import { useSelector } from 'react-redux';
import InventoryList from './InventoryList';
import Cart from './Cart';
import './App.css';
import ItemDetail from './Item/ItemDetail';

function App() {

  return (
    <div className="App">
        <AppHeader />
      <main className="App-main">
        <Switch>
          <Route exact path="/">
            <InventoryList />
          </Route>

          <Route exact path="/cart">
            <Cart />
          </Route>
          
          <Route exact path="/items/:id">
            <ItemDetail />
          </Route>

          <Redirect to="/" />
        </Switch>
      </main>
    </div>
  );
}

function AppHeader () {
  const itemsInCart = useSelector(state => {
    Object.keys(state.cart).reduce((totalItems, key) => (
      totalItems + state.cart[key].qty
    ), 0);
  });
  return (
    <header className="App-header">
      <Link to="/" >
          Home
      </Link>
      <Link to="/cart">
          Cart
          <small>
            ({itemsInCart})
          </small>
      </Link>
    </header>
  )
}
export default App;
