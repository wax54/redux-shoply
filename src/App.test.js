import { render } from '@testing-library/react';
import { renderWithRedux } from './testHelpers';
import App from './App';

test("doesn't Blow Up", () => {
  renderWithRedux(<App />);
});

test("Matches Snapshot", () => {
  const { asFragment } = renderWithRedux(<App />);
  expect(asFragment()).toMatchSnapshot();
});
