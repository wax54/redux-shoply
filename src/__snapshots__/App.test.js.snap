// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <div>
        <h2>
          Cart!
        </h2>
        <h4>
          FYI, there are 0 items in your cart right now!
        </h4>
        <div>
          Get Shopping!
        </div>
      </div>
      <div>
        <h1>
          HELLO, HERE IS OUR INVENTORY!
        </h1>
        <ul>
          <div
            id="47314fa1-ae56-4eae-80be-af6691145951"
          >
            <img
              alt="tv"
              height="200px"
              src="https://images.samsung.com/is/image/samsung/latin_en-hd-j4300ah-un32j4300ahxpa-001-front-indigo-blue"
            />
            <h4>
              tv - $219.99
            </h4>
            <p>
              A beautiful, big-screen TV. Because hey, Netflix isn't going to watch itself.
            </p>
            <div />
            <button>
              Add One To Cart
            </button>
            <button>
              Remove One From Cart
            </button>
          </div>
          <div
            id="3fdd689a-02cc-4ae7-903b-f6d2776ff3b9"
          >
            <img
              alt="microwave"
              height="200px"
              src="https://target.scene7.com/is/image/Target/GUEST_c59b9457-3ee1-463b-ad0a-42f48d06c140"
            />
            <h4>
              microwave - $100
            </h4>
            <p>
              Heat your food with the power of SCIENCE!
            </p>
            <div />
            <button>
              Add One To Cart
            </button>
            <button>
              Remove One From Cart
            </button>
          </div>
          <div
            id="b04b779c-1cfb-48a7-8424-25a8c7e608ae"
          >
            <img
              alt="toaster oven"
              height="200px"
              src="https://images-na.ssl-images-amazon.com/images/I/81110bb7g2L._SL1500_.jpg"
            />
            <h4>
              toaster oven - $20.49
            </h4>
            <p>
              Tasty bread not included.
            </p>
            <div />
            <button>
              Add One To Cart
            </button>
            <button>
              Remove One From Cart
            </button>
          </div>
          <div
            id="4260bf52-218a-480e-87ea-4ff42470ce98"
          >
            <img
              alt="chair"
              height="200px"
              src="https://www.ikea.com/PIAimages/0355482_PE547815_S5.JPG"
            />
            <h4>
              chair - $100.89
            </h4>
            <p>
              It's a chair. You can sit in it.
            </p>
            <div />
            <button>
              Add One To Cart
            </button>
            <button>
              Remove One From Cart
            </button>
          </div>
          <div
            id="6a56c22f-2a85-4751-98a9-14e280fcecd8"
          >
            <img
              alt="mirror"
              height="200px"
              src="https://target.scene7.com/is/image/Target/GUEST_703892ef-2db0-4e02-b9c8-685e4a5c37a0"
            />
            <h4>
              mirror - $20.99
            </h4>
            <p>
              Mirror mirror on the wall, who has the greatest mirror of them all? YOU WILL, if you buy this.
            </p>
            <div />
            <button>
              Add One To Cart
            </button>
            <button>
              Remove One From Cart
            </button>
          </div>
          <div
            id="2266b2d6-3d71-4356-8065-392f1fceab07"
          >
            <img
              alt="phone"
              height="200px"
              src="https://images-na.ssl-images-amazon.com/images/I/717DO5Q-aCL._SY879_.jpg"
            />
            <h4>
              phone - $399.99
            </h4>
            <p>
              Because there are absolutely no downsides that come with easier access to social media.
            </p>
            <div />
            <button>
              Add One To Cart
            </button>
            <button>
              Remove One From Cart
            </button>
          </div>
        </ul>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
